File: problem/Max_Vowels/maxVowels.c
int maxVowels(char *s, int k)
{
    int max_vowels = 0;
    int curr_vowels = 0;
    int left = 0;
    int len = strlen(s);

    for (int i = 0; i < k; i++)
    {
        if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u')
        {
            curr_vowels++;
        }
    }

    max_vowels = curr_vowels;

    for (int i = k; i < len; i++)
    {
        if (s[left] == 'a' || s[left] == 'e' || s[left] == 'i' || s[left] == 'o' || s[left] == 'u')
        {
            curr_vowels--;
        }
        left++;
        if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u')
        {
            curr_vowels++;
        }
        if (curr_vowels > max_vowels)
            max_vowels = curr_vowels;
    }

    return max_vowels;
}

File: problem/Max_Vowels/maxVowels.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {
    let vowels = new Set(['a', 'e', 'i', 'o', 'u']);
    let max_vowels = 0;
    let curr_vowels = 0;
    let left = 0;
    
    for (let i = 0; i < k; i++) {
        if (vowels.has(s[i])) {
            curr_vowels++;
        }
    }
    
    max_vowels = curr_vowels;
    
    for (let i = k; i < s.length; i++) {
        if (vowels.has(s[left])) {
            curr_vowels--;
        }
        left++;
        if (vowels.has(s[i])) {
            curr_vowels++;
        }
        max_vowels = Math.max(max_vowels, curr_vowels);
    }
    
    return max_vowels;
};